DisguiseUnityRenderStream/Runtime/PipelineAbstraction/CameraCapture.cs

```diff
@@ -0,0 +1,275 @@
+        /// </para>
+        ///
+        /// <remarks>
+        /// To avoid unnecessary texture copies, this refers directly to <see cref="Camera.targetTexture"/>.
+        /// </remarks>
+        /// </summary>
+        public RenderTexture cameraTexture => m_cameraTexture;
+        
+        /// <summary>
+        /// The captured depth texture. Its format is defined by <see cref="description"/>.
+        /// </summary>
+        public RenderTexture depthTexture => m_depthTexture;
+
+        /// <summary>
+        /// Defines the color and depth textures. The textures are automatically disposed and created on change.
+        /// </summary>
+        public CameraCaptureDescription description
+        {
+            get => m_description;
+            set
+            {
+                if (m_description != value)
+                {
+                    m_description = value;
+                    Refresh();
+                }
+            }
+        }
+
+        [SerializeField]
+        CameraCaptureDescription m_description = CameraCaptureDescription.Default;
+
+        DepthCopy m_depthCopy;
+        Camera m_camera;
+        RenderTexture m_cameraTexture;
+        RenderTexture m_depthTexture;
+        
+#if DEBUG
+        bool m_HasSetSecondNames;        
+#endif
+
+#if UNITY_EDITOR
+        CameraCaptureDescription m_lastDescription = CameraCaptureDescription.Default;
+        bool m_RefreshFlag; // Ensures thread safety (OnValidate is on a different thread)
+        
+        void OnValidate()
+        {
+            if (m_lastDescription != m_description)
+            {
+                m_lastDescription = m_description;
+                m_RefreshFlag = true;
+            }
+        }
+
+        void Update()
+        {
+            if (m_RefreshFlag)
+            {
+                m_RefreshFlag = false;
+                Refresh();
+            }
+        }
+#endif
+
+        void Awake()
+        {
+            m_depthCopy = new DepthCopy();
+            
+            m_camera = GetComponent<Camera>();
+            
+            if (m_description.IsValid)
+                CreateResources(m_description);
+        }
+
+        void OnEnable()
+        {
+            RenderPipelineManager.endCameraRendering += OnEndCameraRendering;
+
+            if (m_depthCopy == null)
+                m_depthCopy = new DepthCopy(); // Can be lost after domain reload 
+        }
+
+        void OnDisable()
+        {
+            RenderPipelineManager.endCameraRendering -= OnEndCameraRendering;
+        }
+
+        void OnDestroy()
+        {
+            DisposeResources();
+        }
+
+        void Refresh()
+        {
+            DisposeResources();
+            CreateResources(m_description);
+            m_camera.targetTexture = m_cameraTexture;
+        }
+
+        void CreateResources(CameraCaptureDescription desc)
+        {
+            if (!desc.IsValid)
+            {
+                Debug.LogWarning("Invalid description");
```

Comment:
Need a more informative message here, e.g. "Camera capture has invalid description, will be disabled."